### PR TITLE
Refactor decoding Base64 to byte[]

### LINKED ISSUE #9911: [Enhancement]: Introduce new methods in operator-common Util class for decoding Base64 to byte[]
### Related problem

_No response_

### Suggested solution

Introduce 2 new methods:
<code>
public static byte[] decodeBytesFromBase64(String data)
public static byte[] decodeBytesFromBase64(byte[] data)
</code>

and refactor existing classes calling <code>Base64.getDecoder().decode(...)</code> to use the new methods

### Alternatives

_No response_

### Additional context

Creating as a result of comment in PR #9906

### PR DESCRIPTION
**Pull request by MichaelMorrisEst**
### Type of change

Refactoring

### Description

Introduce new methods in operator-common Util class for decoding Base64 to byte[] and refactor code to use the new methods

Closes #9911 

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [X] Make sure all tests pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [X] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

File: systemtest/src/main/java/io/strimzi/systemtest/resources/keycloak/SetupKeycloak.java [hunk 1]
```diff
@@ -127,10 +128,10 @@ private static KeycloakInstance createKeycloakInstance(String namespaceName) {
         Secret keycloakSecret = kubeClient().getSecret(namespaceName, KEYCLOAK_SECRET_NAME);
 
         String usernameEncoded = keycloakSecret.getData().get("username");
-        String username = new String(Base64.getDecoder().decode(usernameEncoded.getBytes(StandardCharsets.UTF_8)), StandardCharsets.UTF_8);
+        String username = new String(Util.decodeBytesFromBase64(usernameEncoded.getBytes(StandardCharsets.UTF_8)), StandardCharsets.UTF_8);
```
Comment (scholzj) [inline systemtest/src/main/java/io/strimzi/systemtest/resources/keycloak/SetupKeycloak.java:131]: Here, but also in the other cases of this ... should these go directly to String instead using the already existing `Utils.decodeFromBase64(String data, Charset charset)` method? I think when the `byte[]` is actually used, `Util.decodeBytesFromBase64` makes sense. but where we just create a String from it right away, we should use the method to decode it directly to String, or?

**Review by scholzj — COMMENTED** [review]
Thanks for the PR ... I left one comment that might apply to more places.

File: systemtest/src/main/java/io/strimzi/systemtest/resources/keycloak/SetupKeycloak.java (hunk 1 above)
Comment (MichaelMorrisEst) [reply systemtest/src/main/java/io/strimzi/systemtest/resources/keycloak/SetupKeycloak.java]: Yes, thats a good point, I've updated the implementation

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/model/CertUtils.java [hunk 2]
```diff
@@ -15,14 +15,13 @@
...
 import java.math.BigInteger;
-import java.nio.charset.StandardCharsets;
 import java.security.cert.CertificateEncodingException;
 import java.util.ArrayList;
-import java.util.Base64;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 
+
```
Comment (scholzj) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/model/CertUtils.java:24]: Not needed I guess.

File: systemtest/src/main/java/io/strimzi/systemtest/resources/keycloak/SetupKeycloak.java [hunk 3]
```diff
@@ -127,10 +128,10 @@ private static KeycloakInstance createKeycloakInstance(String namespaceName) {
         Secret keycloakSecret = kubeClient().getSecret(namespaceName, KEYCLOAK_SECRET_NAME);
 
         String usernameEncoded = keycloakSecret.getData().get("username");
-        String username = new String(Base64.getDecoder().decode(usernameEncoded.getBytes(StandardCharsets.UTF_8)), StandardCharsets.UTF_8);
+        String username = Util.decodeFromBase64(usernameEncoded);
 
         String passwordEncoded = keycloakSecret.getData().get("password");
-        String password = new String(Base64.getDecoder().decode(passwordEncoded.getBytes(StandardCharsets.UTF_8)), StandardCharsets.UTF_8);
+        String password = Util.decodeFromBase64(passwordEncoded);
```
Comment (scholzj) [inline systemtest/src/main/java/io/strimzi/systemtest/resources/keycloak/SetupKeycloak.java:134]: Should these keep the UTF_8 charset?

**Review by scholzj — COMMENTED** [review]
Few more nits. But LGTM overall, thanks.

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/model/CertUtils.java (hunk 2 above)
Comment (MichaelMorrisEst) [reply cluster-operator/src/main/java/io/strimzi/operator/cluster/model/CertUtils.java]: Done

File: systemtest/src/main/java/io/strimzi/systemtest/resources/keycloak/SetupKeycloak.java (hunk 3 above)
Comment (MichaelMorrisEst) [reply systemtest/src/main/java/io/strimzi/systemtest/resources/keycloak/SetupKeycloak.java]: Done

**Review by scholzj — APPROVED** [review]
Thanks. LGTM. I will run the system tests, but I do not expect anything to break there.

**Review by ppatierno — APPROVED** [review]
LGTM. Thanks for the PR!

**Review by fvaleri — APPROVED** [review]
Thanks.

**Pull request merged**